cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -104,6 +104,16 @@ func GetStaticPodSpecs(cfg *kubeadmapi.MasterConfiguration, k8sVersion *version.
 		}, mounts.GetVolumes(kubeadmconstants.KubeScheduler)),
 	}
 
+	if cfg.CloudProvider == "openstack" {
```

Comment:
Is this the first cloud provider specific things kubeadm is doing? I would like to see the plan for this going forward… this is just moving compiled-in logic up a level, when what we really want is to enable external integration of cloud provider specific modifications.

Reply:
@liggitt there's at lease one precedent:

https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/controlplane/manifests.go#L174

Reply:
Interesting. The linked example seems like a bug… admission plugins don't depend solely on cloud provider… seems like specifying cloud provider aws with kube 1.8 would not work with that code